app/src/main/java/com/codepath/apps/restclienttemplate/TweetsAdapter.java

```diff
@@ -145,6 +185,122 @@ public void bind(Tweet tweet) {
 
             tvBody.setTextColor(Color.BLACK);
             tvName.setTextColor(Color.BLACK);
+
+            tvRetweet.setText(String.valueOf(tweet.retweetCount));
+            tvFavorite.setText(String.valueOf(tweet.favoriteCount));
+
+            if (tweet.retweeted) {
+                ivRetweet.setImageResource(R.drawable.ic_vector_retweet);
+                ivRetweet.setTag(R.drawable.ic_vector_retweet);
+            } else {
+                ivRetweet.setImageResource(R.drawable.ic_vector_retweet_stroke);
+                ivRetweet.setTag(R.drawable.ic_vector_retweet_stroke);
+            }
+
+            if (tweet.favorited) {
+                ivFavorite.setImageResource(R.drawable.ic_vector_heart);
+                ivFavorite.setTag(R.drawable.ic_vector_heart);
+            } else {
+                ivFavorite.setImageResource(R.drawable.ic_vector_heart_stroke);
+                ivFavorite.setTag(R.drawable.ic_vector_heart_stroke);
+            }
+
+
+            ivRetweet.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    Tweet tweet = tweets.get(getAdapterPosition());
+
+                    if ((int) ivRetweet.getTag() == R.drawable.ic_vector_retweet_stroke) {
+                        // tweet has not been retweeted yet, so this click must handle a retweet
+                        client.retweet(tweet.id, new JsonHttpResponseHandler() {
+                            @Override
+                            public void onSuccess(int statusCode, Headers headers, JSON json) {
+
+                            }
+
+                            @Override
+                            public void onFailure(int statusCode, Headers headers, String response, Throwable throwable) {
+
```

Comment:
You should handle failure here, and same with onSuccess. API requests don't always work and you really need to know when they fails to fix the issue